benchmark/_cli.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+// Create an object of all benchmark scripts
+const benchmarks = {};
+fs.readdirSync(__dirname)
+  .filter(function(name) {
+    return fs.statSync(path.resolve(__dirname, name)).isDirectory();
+  })
+  .forEach(function(category) {
+    benchmarks[category] = fs.readdirSync(path.resolve(__dirname, category))
+      .filter((filename) => filename[0] !== '.' && filename[0] !== '_');
+  });
+
+function CLI(usage, settings) {
+  if (!(this instanceof CLI)) return new CLI(usage, settings);
+
+  if (process.argv.length < 3) {
+    this.abort(usage);
+    return;
+  }
+
+  this.usage = usage;
+  this.optional = {};
+  this.required = [];
+
+  for (const argName of settings.arrayArgs) {
+    this.optional[argName] = [];
+  }
+
+  let currentOptional = null;
+  let detectOptionals = false;
+
+  for (const arg of process.argv.slice(2)) {
+    // Optional arguments declaration
+    if (arg[0] === '-') {
+      if (arg[1] === '-') {
+        currentOptional = arg.slice(2);
+      } else {
+        currentOptional = arg.slice(1);
+      }
+
+      // There can be no more required arguments
+      detectOptionals = true;
+
+      // Default the value to true
+      if (!settings.arrayArgs.includes(currentOptional)) {
+        this.optional[currentOptional] = true;
+      }
+    }
+    // Optional arguments value
+    else if (currentOptional !== null) {
+      if (settings.arrayArgs.includes(currentOptional)) {
+        this.optional[currentOptional].push(arg);
+      } else {
+        this.optional[currentOptional] = arg;
+      }
+      currentOptional = null;
+    }
+    // Required arguments
+    else if (!detectOptionals) {
```

Comment:
/cc @Trott @silverwind Didn't think the next line `else` was allowed in lint. It's not done anywhere else in core, and only 14 places in tests. Has been a rule, to the best of my knowledge, that has been at least informally enforced. There an option for this in the linter?